lldb/CMakeLists.txt

```diff
@@ -44,6 +44,10 @@ endif()
 
 if (WIN32)
   add_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)
+  if (NOT MSVC)
```

Comment:
Do we want this msvc check?

Our Windows on Arm buildbot is using clang-cl, not sure if it'll hit this path. No harm in landing this and finding out though.

Reply:
clang-cl should be detected as MSVC.  The define should be generally safe, but it is nice to be more precise and add it when needed.

Reply:
Defining `_BSD_SOURCE` is probably not changing anything on our existing clang-cl Windows on Arm build then. This will just remove a define we never needed there, as you said.

Reply:
This was never defined for clang-cl/MSVC builds before. Note that the existing define I'm removing is within this context:

```
#if !defined(_MSC_VER) && !defined(__NetBSD__)

#ifdef _WIN32
#define _BSD_SOURCE // Required so that getopt.h defines optreset
#endif
```
So we previously only defined `_BSD_SOURCE` on `defined(_WIN32) && !defined(_MSC_VER)`, i.e. `if (WIN32 AND NOT MSVC)` in cmake (even if the condition nesting is the other way around there).